nano/secure/ledger.cpp

```diff
@@ -865,6 +865,50 @@ std::optional<nano::pending_info> nano::ledger::pending_info (store::transaction
 	return store.pending.get (transaction, key);
 }
 
+std::deque<std::shared_ptr<nano::block>> nano::ledger::confirm (nano::store::write_transaction const & transaction, nano::block_hash const & hash)
+{
+	std::deque<std::shared_ptr<nano::block>> result;
+	std::deque<nano::block_hash> queue;
+	queue.push_front (hash);
+	while (!queue.empty ())
+	{
+		auto hash = queue.front ();
+		auto block = this->block (transaction, hash);
+		release_assert (block);
+		auto dependents = dependent_blocks (transaction, *block);
+		for (auto const & dependent : dependents)
+		{
+			if (!dependent.is_zero () && !block_confirmed (transaction, dependent))
+			{
+				queue.push_front (dependent);
+			}
+		}
+		if (queue.front () == hash)
+		{
+			queue.pop_front ();
+			if (!block_confirmed (transaction, hash))
+			{
+				result.push_back (block);
+				confirm (transaction, *block);
+			}
+		}
+		else
+		{
+			// unconfirmed dependencies were added
+		}
+	}
+	return result;
+}
+
+void nano::ledger::confirm (nano::store::write_transaction const & transaction, nano::block const & block)
+{
+	debug_assert (!store.confirmation_height.get (transaction, block.account ()) || store.confirmation_height.get (transaction, block.account ()).value ().height < block.sideband ().height);
```

Comment:
Can this check be tightened to assert that current confirmation height is exactly one less than block height. From what I understand this function is supposed to be used to incrementally confirm blocks in account chain, one by one. Is my understanding correct?

Reply:
Yep that's true. Changed.